inc/blocks/broadcast/edit.js

```diff
@@ -0,0 +1,230 @@
+	const [ isFetching, setIsFetching ] = useState( false );
+	const [ searchKeyword, _setSearchKeyword ] = useState( '' );
+
+	const fetchBroadcasts = debouncePromise( function ( keyword, extraArgs ) {
+		const args = {
+			path: addQueryArgs( '/accelerate/v1/broadcast', { search: keyword } ),
+			parse: false,
+			...extraArgs,
+		};
+
+		return apiFetch( args )
+			.then( response => response.json ? response.json() : [] );
+	}, 1000 );
+
+	const fetchBroadcast = debouncePromise( function ( id, extraArgs ) {
+		const args = {
+			path: addQueryArgs( `accelerate/v1/broadcast/${ id }` ),
+			parse: false,
+			...extraArgs,
+		};
+
+		return apiFetch( args ).then( response => response.json ? response.json() : null );
+	}, 1000 );
+
+	/**
+	 * Select a Broadcast block.
+	 *
+	 * @param {*} broadcast Broadcast block metadata
+	 * @returns {void}
+	 */
+	const setBroadcast = function ( broadcast ) {
+		if ( ! broadcast && ! window.confirm( __( 'Are you sure you want to replace this Broadcast ?', 'altis' ) ) ) {
+			return;
+		}
+		_setBroadcast( broadcast );
+		setAttributes( { broadcast: broadcast?.id || null } );
+	};
+
+	/**
+	 * Normalize and set search keyword.
+	 *
+	 * @param {string} keyword Search keyword.
+	 */
+	const setSearchKeyword = function ( keyword ) {
+		// Disregard diacritics.
+		//  Input: "média"
+		keyword = _deburr( keyword );
+
+		// Accommodate leading slash, matching autocomplete expectations.
+		//  Input: "/media"
+		keyword = keyword.replace( /^\//, '' );
+
+		// Strip leading and trailing whitespace.
+		//  Input: " media "
+		keyword = keyword.trim();
+
+		_setSearchKeyword( keyword );
+	};
+
+	// Set the block clientId if none currently set.
+	useEffect( () => {
+		if ( ! attributes.clientId ) {
+			setAttributes( { clientId: uuid() } );
+		}
+	}, [ attributes.clientId, setAttributes ] );
+
+	// Fetch the broadcast post if we have one selected already.
+	useEffect( () => {
+		const abortController = new AbortController();
+		if ( attributes.broadcast && ! broadcast?.id ) {
+			fetchBroadcast( attributes.broadcast, { signal: abortController.signal } ).then( setBroadcast );
+		}
+
+		return () => abortController.abort();
+	}, [ attributes.broadcast ] );
+
+	// Search broadcasts, and load initial broadcast list.
+	useEffect( () => {
+		const abortController = new AbortController();
+		setIsFetching( true );
+		fetchBroadcasts( searchKeyword, { signal: abortController.signal } )
+			.then( setBroadcasts )
+			.catch( () => {
+				setBroadcasts( [] );
+			} )
+			.finally( () => setIsFetching( false ) );
+
+		return () => abortController.abort();
+	}, [ searchKeyword ] );
+
+	// Clear search term when a broadcast is selected.
+	useEffect( () => {
+		if ( broadcast ) {
+			setSearchKeyword( '' );
+		}
+	}, [ broadcast ] );
+
+	return (
+		<Fragment>
+			<InspectorControls>
+				<div className="components-panel__body is-opened">
+					<Button
+						href={ window.Altis.Analytics.Broadcast.ManagerURL }
```

Comment:
Added an icon for clarity.

Reply:
In accessibility terms an icon isn’t enough, think screen reader users. I believe if there are unsaved changes it will prompt you before navigating away anyway. Will have to see what it’s like to use I suppose

Reply:
I also added a title, but perhaps I forgot to say I also added a title.